packages/plugin-react/src/index.ts

```diff
@@ -89,261 +77,198 @@ declare module 'vite' {
 
 const prependReactImportCode = "import React from 'react'; "
 const refreshContentRE = /\$Refresh(?:Reg|Sig)\$\(/
+const defaultIncludeRE = /\.(?:mjs|[tj]sx?)$/
 
 export default function viteReact(opts: Options = {}): PluginOption[] {
   // Provide default values for Rollup compat.
   let devBase = '/'
-  let filter = createFilter(opts.include, opts.exclude)
+  const filter = createFilter(
+    [
+      defaultIncludeRE,
+      ...(Array.isArray(opts.include)
+        ? opts.include
+        : opts.include
+        ? [opts.include]
+        : []),
```

Comment:
Maybe you could copy the helper we use in Vite core so this is easier to follow for others
```
export function arraify<T>(target: T | T[]): T[] {
  return Array.isArray(target) ? target : [target]
}
```